internal/lsp/server.go

```diff
@@ -211,18 +211,14 @@ func (l *LanguageServer) StartDiagnosticsWorker(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case evt := <-l.diagnosticRequestFile:
-			success, err := l.processTextContentUpdate(ctx, evt.URI, evt.Content)
+			err := l.processTextContentUpdate(ctx, evt.URI, evt.Content)
```

Comment:
This has been updated to avoid any special handling of parse errors. diagnostics are always sent from the parent now. This is unrelated to the title fix.